sdk/src/test/java/com/qonversion/android/sdk/internal/userProperties/UserPropertiesServiceImplTest.kt

```diff
@@ -0,0 +1,183 @@
+package com.qonversion.android.sdk.internal.userProperties
+
+import com.qonversion.android.sdk.assertThatQonversionExceptionThrown
+import com.qonversion.android.sdk.internal.common.mappers.UserPropertiesMapper
+import com.qonversion.android.sdk.internal.exception.ErrorCode
+import com.qonversion.android.sdk.internal.logger.Logger
+import com.qonversion.android.sdk.internal.networkLayer.apiInteractor.ApiInteractor
+import com.qonversion.android.sdk.internal.networkLayer.dto.Request
+import com.qonversion.android.sdk.internal.networkLayer.dto.Response
+import com.qonversion.android.sdk.internal.networkLayer.requestConfigurator.RequestConfigurator
+import io.mockk.*
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runTest
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Nested
+
+import org.junit.jupiter.api.Test
+
+internal class UserPropertiesServiceImplTest {
+    private lateinit var service: UserPropertiesServiceImpl
+
+    private val mockRequestConfigurator = mockk<RequestConfigurator>()
+    private val mockApiInteractor = mockk<ApiInteractor>()
+    private val mockMapper = mockk<UserPropertiesMapper>()
+    private val mockLogger = mockk<Logger>()
+
+    private val slotErrorLogMessage = slot<String>()
+
+    @BeforeEach
+    fun setUp() {
+        every {
+            mockLogger.error(capture(slotErrorLogMessage))
+        } just runs
+
+        service = UserPropertiesServiceImpl(
+            mockRequestConfigurator,
+            mockApiInteractor,
+            mockMapper,
+            mockLogger
+        )
+    }
+
+    @ExperimentalCoroutinesApi
+    @Nested
+    inner class SendPropertiesTest {
+        @Test
+        fun `send properties success`() = runTest {
+            // given
+            val spykService = spyk(service)
+
+            val expectedProperties = listOf("_q_email", "_q_adjust_adid")
+            val responseData = mapOf(
+                "result" to "ok",
+                "error" to "",
+                "errors" to listOf<String>(),
+                "processed" to expectedProperties
+            )
```

Comment:
I think response data can be any mock, even just empty string because we don't actually need the real value as we spyk mapper